### PR TITLE
port: tests v10/adapters fetch express fastify

### LINKED ISSUE #3229: feat: port all tests from v9 to v10
## Describe the feature you'd like to request

We have a lot of tests that are outdated and using v9. We should have them all in v10-structure.

Good intro if you are curious about contributing to tRPC.

### Instructions

- https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md
- Rewrite all of https://github.com/trpc/trpc/tree/main/packages/tests/server/interop to a nice v10 equivalent (the v10 tests are in the folder above)
- Some have equivalents already
- They can be done one-by-one
- Improvements on the general testing structure is also welcome

#### Example

1. Copy `/packages/tests/server/interop/outputParser.test.ts` -> `/packages/tests/server/outputParser.test.ts` 
1. Replace `legacyRouterToServerAndClient` with `routerToServerAndClientNew`
1. Rewrite the routers using tRPC 10
1. Rewrite the calls using tRPC 10

### Get paid

- We can pay $5-10 per test file rewritten if done well. More for bigger ones (like subscriptions and WebSockets)
- Once you get into the flow, it might not take more than 5 minutes per test file.

### Full tree

```
├── adapters
│   ├── express.test.tsx
│   ├── fastify.test.ts
│   ├── fetch.test.ts
│   └── next.test.ts
├── clientInternals.test.ts
├── dataloader.test.ts
├── errors.test.ts
├── headers.test.tsx
├── index.test.tsx
├── inference.test.ts
├── links.test.ts
├── middleware.test.ts
├── outputParser.test.ts
├── rawFetch.test.tsx
├── react
│   ├── __testHelpers.tsx
│   ├── dehydrate.test.tsx
│   ├── formatError.test.tsx
│   ├── infiniteQuery.test.tsx
│   ├── invalidateQueries.test.tsx
│   ├── prefetchQuery.test.tsx
│   ├── regression
│   │   └── issue-1645-setErrorStatusSSR.test.tsx
│   ├── setInfiniteQueryData.test.tsx
│   ├── setQueryData.test.tsx
│   ├── useMutation.test.tsx
│   ├── useQuery.test.tsx
│   ├── useSubscription.test.tsx
│   └── withTRPC.test.tsx
├── regression
│   ├── issue-949-inferProcedureInput.test.ts
│   └── issue-990-initialData.test.tsx
├── responseMeta.test.ts
├── router.test.ts
├── routerMeta.test.ts
├── transformer.test.ts
├── validators.test.ts
```

### PR DESCRIPTION
**Pull request by nilskj**
Closes #3229 

## 🎯 Changes

Port of Express and fastify adapter tests with helper __router to v10.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/tests/server/adapters/fetch.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/**
+ * @jest-environment miniflare
+ */
+/// <reference types="@cloudflare/workers-types" />
+import { Context, router } from './__router';
+import { Response as MiniflareResponse } from '@miniflare/core';
+import {
```
Comment (KATT) [inline packages/tests/server/adapters/fetch.test.ts:7]: lint fix needed

**Review by KATT — APPROVED** [review]

File: packages/tests/server/adapters/fetch.test.ts (hunk 1 above)
Comment (nilskj) [reply packages/tests/server/adapters/fetch.test.ts]: done!

**Comment by nilskj** [comment]
I accidently pushed fastify even though It's not done yet, wait before merging

**Comment by nilskj** [comment]
I have ported fastify adapter tests as well now! @KATT there are some lambda tests already in target folder that still use .interop(), I guess these should be ported too right? I will have a look tonight if no one else snags them. All adapters are done except these 👍

**Review by KATT — APPROVED** [review]

**Pull request merged**